packages/http/src/mocker/__tests__/HttpMocker.spec.ts

```diff
@@ -229,6 +228,100 @@ describe('HttpMocker', () => {
+          });
+        });
+
+        describe('and the resource has not an example', () => {
+          const resourceOperation: IHttpOperation = {
+            id: 'id',
+            method: 'get',
+            path: '/test',
+            request: {},
+            responses: [
+              {
+                code: '200',
+                headers: [],
+                contents: [
+                  {
+                    mediaType: 'application/json',
+                    schema: {
+                      type: 'object',
+                      properties: {
+                        name: { type: 'string', examples: ['Clark'] },
+                        middlename: { type: 'string', examples: ['J'], default: 'JJ' },
+                        surname: { type: 'string', default: 'Kent' },
+                        age: { type: ['number', 'null'] },
+                        email: { type: 'string' },
+                        deposit: { type: 'number' },
+                        pet: {
+                          type: 'object',
+                          properties: {
+                            name: { type: 'string', examples: ['Clark'] },
+                            middlename: { type: 'string', examples: ['J'], default: 'JJ' },
+                          },
+                        },
+                      },
+                      required: ['name', 'surname', 'age', 'email'],
+                    },
+                    encodings: [],
+                  },
+                ],
+              },
+            ],
+          };
+
+          const response = httpMocker.mock({
+            input: mockInput,
+            resource: resourceOperation,
+            config: { mock: { dynamic: false } },
+          });
+
+          describe('the property has an example key', () => {
+            it('should return the example key', () => expect(response.body).toHaveProperty('name', 'Clark'));
+            describe('the property has an example and a default key', () => {
+              it('should still prefer the example', () => expect(response.body).toHaveProperty('middlename', 'J'));
+            });
+
+            describe('with a objects', () => {
+              it('should return the example key', () => expect(response.body).toHaveProperty('pet.name', 'Clark'));
+              it('should still prefer the example', () => expect(response.body).toHaveProperty('pet.middlename', 'J'));
+            });
+          });
+
+          describe('the property has not an example, but a default key', () => {
+            it('should use such key', () => {
+              expect(response.body).toHaveProperty('surname', 'Kent');
+            });
+          });
+
+          describe('the property has nor default, nor example', () => {
+            describe('is nullable', () => {
+              it('should be set to null', () => expect(response.body).toHaveProperty('age', null));
+            });
+
+            describe('and is not nullable', () => {
```

Comment:
What about if it is a non null enum string. 

What about if the property is oneOf/allOf/anyOf what does it show.

Reply:
Does it support $ref

Reply:
> What about if it is a non null enum string.

Returns the first enum value

> oneOf

Returns the first value

> allOf

Merges and returns (as long they're compatible)

> anyOf

Returns the first value

> $ref

Apparently not supported.

I'm writing the tests for this as we speak.

Reply:
@philsturgeon You have them now.

https://github.com/stoplightio/prism/pull/347/commits/9872b0a09cc50ec912cb7c07a1eba55d89a117bc